CMakeLists.txt

```diff
@@ -1,16 +1,24 @@
 cmake_minimum_required(VERSION 3.5)
 project(LibreCuda CXX)
 
-# create elf-parser library
-add_subdirectory(third_party/ELFIO)
+set( CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin )
+set( CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib )
+
+include_directories(
```

Comment:
Please always use target_include_directories to be precise. Maybe also separate src/ and headers into include/ and internal/ headers?
Headers currently have an _internal.h postfix, but it is intended for future separation to not be included automatically if you link against the module.
Also, no need for include_directories, linking against a module that has headers publicly included will automatically result in inclusion to the target module.

TLDR:
dont include_directories()

Instead:
target_linclude_directories(driverapi PRIVATE internal/)
target_include_directories(driverapi PUBLIC include/)

Then
with target_link_libraries(write_float_test PRIVATE driverapi), write_float_test will have access to headers anyway